test/jdk/java/beans/Introspector/DefaultMethodBeanPropertyTest.java

```diff
@@ -64,30 +77,135 @@ public Float getObj() {
     }
 
-    public static void main(String[] args) throws Exception {
-        findProperty(ClassB.class, "foo");
+    public interface B2 extends A2 {
+    }
 
-        // Expected properties
+    public interface C2 extends A2 {
+    }
+
+    public class D2 implements B2, C2 {
+    }
+
+    public static void testScenario2() {
+        verifyProperties(D2.class,
+            "getClass",
+            "getFoo"
+        );
+    }
+
+//////////////////////////////////////
+//                                  //
+//          SCENARIO 3              //
+//                                  //
+//////////////////////////////////////
+
+    public interface A3 {
+        default Object getFoo() {
+            return null;
+        }
+    }
+
+    public interface B3 extends A3 {
+        @Override
+        Set<?> getFoo();
+    }
+
+    public interface C3 extends A3 {
+        @Override
+        Collection<?> getFoo();
+    }
+
+    public class D3 implements B3, C3 {
+        @Override
+        public NavigableSet<?> getFoo() {
+            return null;
+        }
+    }
+
+    public static void testScenario3() {
+        verifyProperties(D3.class,
+            "getClass",
+            "getFoo"
+        );
+    }
+
+// Helper methods
+
+    public static void verifyProperties(Class<?> type, String... getterNames) {
+
+        // Gather expected properties
         final HashSet<PropertyDescriptor> expected = new HashSet<>();
-        expected.add(new PropertyDescriptor("class", ClassB.class, "getClass", null));  // inherited method
-        expected.add(new PropertyDescriptor("value", ClassB.class, "getValue", null));  // inherited default method
-        expected.add(new PropertyDescriptor("foo", ClassB.class, "getFoo", null));      // overridden interface method
-        expected.add(new PropertyDescriptor("obj", ClassB.class, "getObj", null));      // overridden default method
+        for (String methodName : getterNames) {
+            final String suffix = methodName.substring(3);
+            final String propName = Introspector.decapitalize(suffix);
+            final Method getter;
+            try {
+                getter = type.getMethod(methodName);
+            } catch (NoSuchMethodException e) {
+                throw new Error("unexpected error", e);
+            }
+            final PropertyDescriptor propDesc;
+            try {
+                propDesc = new PropertyDescriptor(propName, getter, null);
+            } catch (IntrospectionException e) {
+                throw new Error("unexpected error", e);
+            }
+            expected.add(propDesc);
+        }
+
+        // Verify properties can be found directly
+        expected.stream()
+          .map(PropertyDescriptor::getName)
+          .filter(name -> BeanUtils.getPropertyDescriptor(type, name) == null)
+          .findFirst()
+          .ifPresent(name -> {
+            throw new Error("property \"" + name + "\" not found in " + type);
+          });
```

Comment:
```suggestion
        expected.stream()
                .map(PropertyDescriptor::getName)
                .filter(name -> BeanUtils.getPropertyDescriptor(type, name) == null)
                .findFirst()
                .ifPresent(name -> {
                    throw new Error("property \"" + name + "\" not found in " + type);
                });
```
I don't know if there's a common style that's followed in OpenJDK, however, aligning wrapped method calls in streams on the dot is the most common way. I prefer this style.

Indenting wrapped lines by two spaces doesn't look standard to me, the most common indentation for wrapped lines seems to be 8 spaces.

Reply:
Thanks - fixed in b92726a923b.